util/vendor.py

```diff
@@ -340,79 +352,94 @@ def main(argv):
         raise SystemExit(sys.exc_info()[1])
     desc['_base_dir'] = vendor_file_base_dir
 
+
+    desc_file_stem = desc_file_path.name.rsplit('.', 2)[0]
+    lock_file_path = desc_file_path.with_name(desc_file_stem + '.lock.hjson')
+
+    # Importing may use lock file upstream, information, so make a copy now
+    # which we can overwrite with the upstream information from the lock file.
+    import_desc = desc.copy()
+
     # Load lock file contents (if possible)
-    desc_file_path_str = str(desc_file_path)
-    lock_file_path = Path(
-        desc_file_path_str[:desc_file_path_str.find('.vendor.hjson')] +
-        '.lock.hjson')
-    try:
-        with open(lock_file_path, 'r') as f:
+    if lock_file_path.is_file():
+        with open(str(lock_file_path), 'r') as f:
             lock = hjson.loads(f.read(), use_decimal=True)
-    except FileNotFoundError:
-        log.warning(
-            "Unable to read lock file %s. Assuming this is the first import.",
-            lock_file_path)
+
+        # Use lock file information for import
+        if not args.update:
+            import_desc['upstream'] = lock['upstream'].copy()
+    else:
         lock = None
+        if not args.update:
+            log.warning("Updating upstream repo as lock file %s not found.",
+                        str(lock_file_path))
+            args.update = True
```

Comment:
nit: I wouldn't modify `args.update`, but use a new variable for it.

Reply:
I think using `args.update` is easier at the moment. We can always revisit this.